### PR TITLE
feat(cmock): Enable linux target build for HID, CDC-ACM and UVC hosts

### PR DESCRIPTION
**Pull request by peter-marcisovsky**
## Description
As a part of Linux host testing, we have created a mock of USB component in the esp-idf, which will be used for testing of Class drivers within this (esp-usb) repository using Cmock test cases. 
 
#### This MR adds Linux target build for:
 - HID Host
 - UVC Host

#### Already buildable Class drivers for linux:
 - CDC ACM Host

#### Not buildable Class drivers for linux:
 - MSC Host - `vfs` not buildable for linux
 - UAC Host - Ringbuffer support missing in Freerots Cmock
 - esp_tinyusb - `vfs` not buildable for linux

As discussed, rather than adding new supported target into the `idf_component.yml` file, (`linux` in this case, and other targets in future) the target list was removed from (those class drivers buildable for linux) and a target compatibility will be handled during build.

## Changes
 - HID Host linux build + simple cmock unit test
 - UVC Host linux build + simple cmock unit test
 - CDC ACM simple cmock unit test

## Related
 -  IDF-10351 - USB host testing: CMock on Linux

## Testing
 - For each class driver buildable on Linux a simple cmock host test has been added

---

## Checklist

Before submitting a Pull Request, please ensure the following:

- [x] 🚨 This PR does not introduce breaking changes.
- [x] All CI checks (GH Actions) pass.
- [x] Documentation is updated as needed.
- [x] Tests are updated or added as necessary.
- [x] Code is well-commented, especially in complex areas.
- [x] Git history is clean — commits are squashed to the minimum necessary.

### CODE REVIEW

**Comment by peter-marcisovsky** [comment]
MR on hold, until idf core team finds a workaround for linux build without using pthread

**Comment by peter-marcisovsky** [comment]
Waiting for esp-idf GL sync wit GH, to enable linux build with the current critical section API.

File: host/class/hid/usb_host_hid/hid_host.c
Comment (roma-jam) [inline host/class/hid/usb_host_hid/hid_host.c:832]:
```suggestion
 * @param[in] iface       Pointer to HID Interface configuration structure
```

File: host/class/hid/usb_host_hid/idf_component.yml
Comment (roma-jam) [inline host/class/hid/usb_host_hid/idf_component.yml:7]: Why we removed targets from here?

**Review by roma-jam — APPROVED** [review]
Cool! That is a huge start, thanks for this! 

Two more questions, otherwise LGTM!

File: host/class/hid/usb_host_hid/idf_component.yml
Comment (peter-marcisovsky) [reply host/class/hid/usb_host_hid/idf_component.yml:7]: I put an explanation to the MR description:

As discussed, rather than adding new supported target into the idf_component.yml file, (linux in this case, and other targets in future) the target list was removed from (those class drivers buildable for linux) and a target compatibility will be handled during build.

So we don't have to update the `idf_component.yml` file for each new target.

**Review by tore-espressif — APPROVED** [review]
LGTM, Thank you !

File: host/class/hid/usb_host_hid/hid_host.c
Comment (peter-marcisovsky) [reply host/class/hid/usb_host_hid/hid_host.c]: Now fixed, also couple of others like this fixed as well.

**Pull request merged**
